target/linux/octeon/patches-6.6/700-allocate_interface_by_label.patch

```diff
@@ -0,0 +1,37 @@
+From: Roman Kuzmitskii <damex.pp@icloud.com>
+Date: Wed, 28 Oct 2020 19:00:00 +0000
+Subject: [PATCH] staging: octeon: add net-labels support
+
+With this patch, device name can be set within dts file
+in the same way as dsa port can.
+
+Add label to pip interface node to use this feature:
+label = "lan0";
+
+Tested-by: Johannes Kimmel <fff@bareminimum.eu>
+Signed-off-by: Roman Kuzmitskii <damex.pp@icloud.com>
+--- a/drivers/staging/octeon/ethernet.c
++++ b/drivers/staging/octeon/ethernet.c
+@@ -407,8 +407,12 @@ static int cvm_oct_common_set_mac_addres
+ int cvm_oct_common_init(struct net_device *dev)
+ {
+ 	struct octeon_ethernet *priv = netdev_priv(dev);
++	const u8 *label = NULL;
+ 	int ret;
+ 
++	if (priv->of_node)
++		label = of_get_property(priv->of_node, "label", NULL);
```

Comment:
Can we drop this whole patch maybe in favor of solving this in userspace, similar to https://github.com/openwrt/openwrt/pull/15593

Reply:
I tried. It's easily doable if the labels are are located in `soc@0/mdio@1180000001800/ethernet-phy@n` (in the specific case of the USG), but for all other devices they're located in `soc@0/pip@11800a0000000/interface@n/ethernet@n`. In the latter case, I couldn't find a trivial way to obtain the device which matched the label. In the former case, it's a script almost identical to yours. If anyone else would like to give it a shot, please do, but there's another elephant in the room (Itus Shield, see below).